src/main/java/net/minecraftforge/unification/UnificationRegistry.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.unification;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.ForgeModContainer;
+import net.minecraftforge.oredict.ShapedOreRecipe;
+import net.minecraftforge.oredict.ShapelessOreRecipe;
+
+/**
+ * The {@link UnificationRegistry} allows mods to register *basic* materials so they can be unified.
+ * <p>
+ * Using the {@link UnificationRegistry}, everything can return a single unified type,
+ * so for example players will only ever get one kind of Copper Ingot or Tin Ore.
+ * <p>
+ * Basic materials follow vanilla conventions, like iron ore being smeltable into iron ingots.
+ * If other mods cannot make basic assumptions about your materials, do not register them for unification.
```

Comment:
Might want to expand on what 'basic assumptions' means. As I understand from the conversation that the system is designed for 100% equal items to be unified. However, the javadocs do not reflect this concept in an intuitive way.

Reply:
Thanks for the feedback, I'll do that.

Reply:
"Nether ores", etc. I guess.

Reply:
Nether ores are an interesting case. I would register them as a new UnificationForm called "nether_ore" instead of "ore".